app/source/main.cpp

```diff
@@ -0,0 +1,321 @@
+	C2D_SpriteSetPos(&go_back, 0, 214);
+	C2D_SpriteSetPos(&header, 95, 14);
+	C2D_SpriteSetPos(&pretendo_loaded_selected, 49, 59);
+	C2D_SpriteSetPos(&pretendo_unloaded_selected, 49, 59);
+	C2D_SpriteSetPos(&pretendo_unloaded_deselected, 49, 59);
+	C2D_SpriteSetPos(&pretendo_loaded_deselected, 49, 59);
+	C2D_SpriteSetPos(&nintendo_loaded_selected, 165, 59);
+	C2D_SpriteSetPos(&nintendo_unloaded_selected, 165, 59);
+	C2D_SpriteSetPos(&nintendo_unloaded_deselected, 165, 59);
+	C2D_SpriteSetPos(&nintendo_loaded_deselected, 165, 59);
+	
+	textBuf = C2D_TextBufNew(4096); // initialize the text buffer with a max glyph count of 4096
+}
+
+/**
+ * Switches the friends and act accounts.
+ *
+ * This will also create the act account if it's missing, just to be safe.
+ * This will fail if the friend account hasn't been created yet.
+ */
+Result switchAccounts(Account friend_account_id) {
+	Result rc = 0;
+
+	handleResult(FRDA_SetLocalAccountId(friend_account_id), "Switch account");
+	if (rc) {
+		return rc;
+	}
+
+	currentEnv = friend_account_id;
+	environment = AccountToNascEnvironment(currentEnv);
+	
+	buttonSelected = currentEnv;
+
+	u32 act_account_index = 0;
+	handleResult(ACTA_GetAccountIndexOfFriendAccountId(&act_account_index, friend_account_id), "Get persistent id for creation");
+
+	if (act_account_index == 0) {
+		handleResult(ACTA_CreateLocalAccount(), "Create act account");
+		handleResult(ACTA_GetAccountIndexOfFriendAccountId(&act_account_index, friend_account_id),
+								 "Get persistent id after creation");
+	}
+
+	handleResult(ACTA_SetDefaultAccount(act_account_index), "Set default account");
+
+	return rc;
+}
+
+Result createAccount(Account friend_account_id) {
+	Result rc = 0;
+
+	// (Re)Create the friend account
+	handleResult(FRDA_CreateLocalAccount(friend_account_id, environment, 0, 1), "Create account");
+	// Switch to the friend/act accounts
+	handleResult(switchAccounts(friend_account_id), "Switch account");
+	// Reset the act account
+	handleResult(ACTA_ResetAccount(friend_account_id, true), "Reset account");
+
+	return rc;
+}
+
+void drawLumaInfo() {
+	DrawString(0.5f, defaultColor, std::format("Luma version is {}.{}.{}\nLuma config version is {}.{}\n\nLuma3DS config bits are:\n{:016b}\n{:016b}\n{:016b}\n{:016b}",
+	std::get<0>(lumaVersion), std::get<1>(lumaVersion), std::get<2>(lumaVersion), std::get<0>(lumaConfigVersion), std::get<1>(lumaConfigVersion), lumaOptions >> 48, (lumaOptions >> 32) & 0xFFFF, (lumaOptions >> 16) & 0xFFFF, lumaOptions & 0xFFFF), 0);
+}
+
+bool checkIfLumaOptionsEnabled(C3D_RenderTarget* top_screen, C3D_RenderTarget* bottom_screen, u32 kDown, u32 kHeld, touchPosition touch) 
+{	
+	kDown |= kHeld; // make kDown have held keys aswell
+
+	C2D_SceneBegin(top_screen);
+	DrawVersionString();
+	
+	C2D_SceneBegin(bottom_screen);
+	
+	// if this is the first time the function has been run, get luma information
+	if (firstStateRun) {
+		firmwareVersion = GetSystemInfoField(0x10000, 0);
```

Comment:
can you add some constants for these parameters?

Reply:
wdym by that?

Reply:
I mean adding constants for values like `0x10000` with what they mean

Reply:
ohhhhhhhhh, ok, gotcha

Reply:
done

Reply:
this also makes it easier to update later when luma changes its config bit/system info field layout next